cmd/flightctl-periodic/main.go

```diff
@@ -2,24 +2,39 @@ package main
 
 import (
 	"context"
+	"time"
 
 	"github.com/flightctl/flightctl/internal/config"
 	"github.com/flightctl/flightctl/internal/instrumentation/tracing"
 	periodic "github.com/flightctl/flightctl/internal/periodic_checker"
 	"github.com/flightctl/flightctl/internal/store"
 	"github.com/flightctl/flightctl/pkg/log"
+	"github.com/flightctl/flightctl/pkg/shutdown"
 	"github.com/sirupsen/logrus"
 )
 
 func main() {
-	ctx := context.Background()
-
 	log := log.InitLogs()
-	log.Println("Starting periodic")
+
+	if err := runCmd(log); err != nil {
+		log.Fatalf("Periodic service error: %v", err)
+	}
+}
+
+func runCmd(log *logrus.Logger) error {
+	log.Info("Starting periodic service")
+	defer log.Info("Periodic service stopped")
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	serverCtx, serverCancel := context.WithCancel(ctx)
```

Comment:
Why is there a need for 2 context.WithCancel ?

Reply:
1. ctx (Main Application Context):
    - Controls the overall application lifecycle
    - Used for coordination between different parts of the application
    - Gets cancelled last in the shutdown sequence
2. serverCtx (Server Context):
    - Child context specifically for server operations
    - Gets cancelled first during graceful shutdown
    - Allows servers to stop accepting new requests while cleanup continues

  // During graceful shutdown:
  serverCancel()  // 1. Stop servers first - no new requests
  // ... allow in-flight requests to complete
  cancel()        // 2. Cancel main context for final cleanup

Reply:
Anyway, can you describe a use case that explains why we need both contexts ?

Reply:
oops, that's a leftover from the older code. Removed it.